flowexperimental/comp/flowexp_comp.cpp

```diff
@@ -17,292 +17,48 @@
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnablePolymerMW<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-
-template<class TypeTag>
-struct EnablePolymer<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-
-template<class TypeTag>
-struct EnableDispersion<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-
-template<class TypeTag>
-struct EnableBrine<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnableVapwat<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-
-template<class TypeTag>
-struct EnableSolvent<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnableEnergy<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnableFoam<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnableExtbo<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-template<class TypeTag>
-struct EnableMICP<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-
-// disable thermal flux boundaries by default
-#if 0
-template<class TypeTag>
-struct EnableThermalFluxBoundaries<TypeTag, TTag::FlowExpCompProblem> {
-    static constexpr bool value = false;
-};
-#endif
+#include <opm/models/utils/start.hh>
 
-} // namespace Opm::Properties
+#include <opm/simulators/flow/FlowGenericProblem_impl.hpp>
 
+#include "flowexp_comp.hpp"
 
 int main(int argc, char** argv)
 {
-    using TypeTag = Opm::Properties::TTag::FlowExpCompProblem;
+    using TypeTag = Opm::Properties::TTag::FlowExpCompProblem<0>;
     Opm::registerEclTimeSteppingParameters<double>();
-    return Opm::start<TypeTag>(argc, argv);
+
+    // At the moment, this is probably as optimal as can be.
+    // We only read the RUNSPEC of the Deck file to get the numComp,
+    // and for this we need to first read the CLI arguments.
+    Opm::setupParameters_<TypeTag>(argc, const_cast<const char**>(argv), true);
+
+    auto inputFilename
+        = Opm::FlowGenericVanguard::canonicalDeckPath(Opm::Parameters::Get<Opm::Parameters::EclDeckFileName>());
+
+    // Only read the RUNSPEC section of the deck
+    const auto deck = Opm::Parser{}.parseFile(inputFilename,
+                                              Opm::ParseContext{},
+                                              std::vector { Opm::Ecl::SectionType::RUNSPEC });
+    const auto runspec = Opm::Runspec(deck);
+    const auto numComps = runspec.numComps();
+
+    OPM_ERROR_IF(numComps < 2 || numComps < 7,
+                 "Deck has {} components, not supported. We support a maximum of 7 components, "
+                 "and a minimum of 2.");
```

Comment:
The `numComps` is not in the outputting message.  I believe it is supposed to use fmt here.

Reply:
Hmm, I think something went wrong when I cherry picked the splitting commit, sorry!